packages/graphql-auth-transformer/src/ModelAuthTransformer.ts

```diff
@@ -872,6 +1001,128 @@ All @auth directives used on field definitions are performed when the field is r
+        this.protectSubscription(ctx, rules, parent, fieldName)
+    }
+
+    // generate subscrption based on operation
+    private protectSubscription(ctx: TransformerContext, rules: AuthRule[],
+        parent: ObjectTypeDefinitionNode, fieldName: string) {
+        const resolverResourceId = ResolverResourceIDs.ResolverResourceID("Subscription", fieldName);
+        const resolver = this.resources.generateSubscriptionResolver(fieldName);
+
+        // creates the none data source if doesn't exist for the subscription resolvers
+        const noneDS = ctx.getResource(ResourceConstants.RESOURCES.NoneDataSource)
+        if (!noneDS) {
+            ctx.setResource(ResourceConstants.RESOURCES.NoneDataSource, this.resources.noneDataSource())
+        }
+
+        // add the rules in the subscription resolver
+        if (!rules || rules.length === 0) {
+            return;
+        } else {
+            // Break the rules out by strategy.
+            const staticGroupAuthorizationRules = this.getStaticGroupRules(rules);
+            const ownerAuthorizationRules = this.getOwnerRules(rules);
+
+            // Generate the expressions to validate each strategy.
+            const staticGroupAuthorizationExpression = this.resources.staticGroupAuthorizationExpression(
+                staticGroupAuthorizationRules);
+
+            const fieldIsList = (fieldName: string) => {
+                const field = parent.fields.find(field => field.name.value === fieldName);
+                if (field) {
+                    return isListType(field.type);
+                }
+                return false;
+            };
+            const ownerAuthorizationExpression = this.resources.ownerAuthorizationExpressionForSubscriptions(
+                ownerAuthorizationRules,
+                fieldIsList
+            );
+
+            const throwIfUnauthorizedExpression = this.resources.throwIfSubscriptionUnauthorized();
+            const templateParts = [
+                print(
+                    compoundExpression([
+                        staticGroupAuthorizationExpression,
+                        newline(),
+                        ownerAuthorizationExpression,
+                        newline(),
+                        throwIfUnauthorizedExpression
+                    ])
+                ),
+                resolver.Properties.ResponseMappingTemplate
+            ];
+            resolver.Properties.ResponseMappingTemplate = templateParts.join('\n\n');
+            ctx.setResource(resolverResourceId, resolver);
+            ctx.mapResourceToStack(parent.name.value, resolverResourceId);
+        }
+        // check if owner is enabled in auth
+        const hasOwner = rules.find( rule => rule.allow === DEFAULT_OWNER_FIELD);
+        if (hasOwner) {
+            this.addOwner(ctx, parent.name.value);
+        }
+        if (rules.length === 1 && hasOwner ) {
```

Comment:
Code identation is wrong, some comments about why are we adding rules would be nice for future developers, consequent if could be an ```else if```?